MAVProxy/modules/lib/mp_util.py

```diff
@@ -250,12 +250,20 @@ def download_url(url):
     if sys.version_info.major < 3:
         from urllib2 import urlopen as url_open
         from urllib2 import URLError as url_error
+        from urllib2 import ProxyHandler as proxy_handler
     else:
         from urllib.request import urlopen as url_open
         from urllib.error import URLError as url_error
+        from urllib.proxy_handler import ProxyHandler as proxy_handler
     try:
```

Comment:
So... in the case that we succeed we drop through and try fetching from the proxy as well.  I'd call that sub-optimal.